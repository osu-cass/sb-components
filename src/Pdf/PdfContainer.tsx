import * as React from 'react';
import { ItemGroupModel } from "./PdfModels";
import { ItemViewContainer } from './ItemViewContainer';
import { CoverPage } from "./CoverPage";


export interface PdfContainerProps {
    items: ItemGroupModel[];
    grade: string;
    subject: string;
    ivsBaseUrl: string;
    cssUrl: string;
    displayTitlePage: boolean;
}

export class PdfContainer extends React.Component<PdfContainerProps, {}> {
    render() {
        const itemPages = this.props.items.map(item => (
            <ItemViewContainer itemData={item} key={item.questions[0].id}/>
        ));

        return (
            <html>
                <head>
                    <meta charSet="UTF-8" />
                    <link rel='stylesheet' href={this.props.cssUrl} />
                    <link rel='stylesheet' href={this.props.ivsBaseUrl + "/Shared/CSS/Universal/items.css"} />
                    <link rel='stylesheet' href={this.props.ivsBaseUrl + "/Projects/SBAC/css/modernItems.css"} />
                </head>
                <body>
                    {this.props.displayTitlePage
                        ? <CoverPage grade={this.props.grade} subject={this.props.subject} />
                        : null}
                    {itemPages}
                </body>
            </html>
        );
    }
}
