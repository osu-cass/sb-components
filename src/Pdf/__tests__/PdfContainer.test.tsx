import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { PdfContainer } from '../PdfContainer';
import { pdfContainerProps } from './PdfMocks';
import { shallow, mount, render } from 'enzyme';

describe("PdfComponent", () => {
    it("matches snapshot", () => {
        expect(shallow(<PdfContainer {...pdfContainerProps} />)).toMatchSnapshot();
    })
})