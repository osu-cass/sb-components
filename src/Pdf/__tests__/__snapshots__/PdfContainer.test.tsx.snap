// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfComponent matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PdfContainer
    cssUrl="http://example.com/style.css"
    displayTitlePage={true}
    grade="3"
    items={
      Array [
        Object {
          "passage": null,
          "questions": Array [
            Object {
              "data": Object {
                "associatedItems": "string",
                "commonCoreStandardsDescription": "string",
                "depthOfKnowledge": "string",
                "educationalDifficulty": "string",
                "evidenceStatement": "string",
                "itemCardViewModel": Object {
                  "bankKey": 1,
                  "claimCode": "234536",
                  "claimLabel": "Math Claim",
                  "commonCoreStandardId": "",
                  "depthOfKnowledge": "",
                  "domain": "MATH",
                  "grade": 8,
                  "gradeLabel": "grade 3",
                  "interactionTypeCode": "",
                  "interactionTypeLabel": "",
                  "isPerformanceItem": false,
                  "itemKey": 1,
                  "subjectCode": "MATH",
                  "subjectLabel": "math",
                  "targetHash": 212,
                  "targetId": "23463467",
                  "targetShortName": "1-3",
                  "title": "",
                },
                "rubrics": Array [
                  Object {
                    "language": "english",
                    "rubricEntries": Array [
                      Object {
                        "name": "",
                        "scorepoint": "",
                        "value": "",
                      },
                    ],
                    "samples": Array [
                      Object {
                        "maxValue": "",
                        "minValue": "",
                        "sampleResponses": Array [
                          Object {
                            "name": "",
                            "purpose": "",
                            "sampleContent": "",
                            "scorePoint": "",
                          },
                        ],
                      },
                    ],
                  },
                ],
                "targetDescription": "string",
              },
              "id": "12334",
              "questionNumber": 1,
              "view": Object {
                "captured": true,
                "html": null,
                "id": "12334",
                "picturePath": null,
                "type": 1,
              },
            },
          ],
        },
      ]
    }
    ivsBaseUrl="http://example.com"
    subject="math"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <head>
          <meta
            charSet="UTF-8"
          />
          <link
            href="http://example.com/style.css"
            rel="stylesheet"
          />
          <link
            href="http://example.com/Shared/CSS/Universal/items.css"
            rel="stylesheet"
          />
          <link
            href="http://example.com/Projects/SBAC/css/modernItems.css"
            rel="stylesheet"
          />
        </head>,
        <body>
          <CoverPage
            grade="3"
            subject="math"
          />
          <ItemViewContainer
            itemData={
              Object {
                "passage": null,
                "questions": Array [
                  Object {
                    "data": Object {
                      "associatedItems": "string",
                      "commonCoreStandardsDescription": "string",
                      "depthOfKnowledge": "string",
                      "educationalDifficulty": "string",
                      "evidenceStatement": "string",
                      "itemCardViewModel": Object {
                        "bankKey": 1,
                        "claimCode": "234536",
                        "claimLabel": "Math Claim",
                        "commonCoreStandardId": "",
                        "depthOfKnowledge": "",
                        "domain": "MATH",
                        "grade": 8,
                        "gradeLabel": "grade 3",
                        "interactionTypeCode": "",
                        "interactionTypeLabel": "",
                        "isPerformanceItem": false,
                        "itemKey": 1,
                        "subjectCode": "MATH",
                        "subjectLabel": "math",
                        "targetHash": 212,
                        "targetId": "23463467",
                        "targetShortName": "1-3",
                        "title": "",
                      },
                      "rubrics": Array [
                        Object {
                          "language": "english",
                          "rubricEntries": Array [
                            Object {
                              "name": "",
                              "scorepoint": "",
                              "value": "",
                            },
                          ],
                          "samples": Array [
                            Object {
                              "maxValue": "",
                              "minValue": "",
                              "sampleResponses": Array [
                                Object {
                                  "name": "",
                                  "purpose": "",
                                  "sampleContent": "",
                                  "scorePoint": "",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                      "targetDescription": "string",
                    },
                    "id": "12334",
                    "questionNumber": 1,
                    "view": Object {
                      "captured": true,
                      "html": null,
                      "id": "12334",
                      "picturePath": null,
                      "type": 1,
                    },
                  },
                ],
              }
            }
          />
        </body>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <meta
              charSet="UTF-8"
            />,
            <link
              href="http://example.com/style.css"
              rel="stylesheet"
            />,
            <link
              href="http://example.com/Shared/CSS/Universal/items.css"
              rel="stylesheet"
            />,
            <link
              href="http://example.com/Projects/SBAC/css/modernItems.css"
              rel="stylesheet"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "charSet": "UTF-8",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "http://example.com/style.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "http://example.com/Shared/CSS/Universal/items.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "http://example.com/Projects/SBAC/css/modernItems.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
        ],
        "type": "head",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CoverPage
              grade="3"
              subject="math"
            />,
            Array [
              <ItemViewContainer
                itemData={
                  Object {
                    "passage": null,
                    "questions": Array [
                      Object {
                        "data": Object {
                          "associatedItems": "string",
                          "commonCoreStandardsDescription": "string",
                          "depthOfKnowledge": "string",
                          "educationalDifficulty": "string",
                          "evidenceStatement": "string",
                          "itemCardViewModel": Object {
                            "bankKey": 1,
                            "claimCode": "234536",
                            "claimLabel": "Math Claim",
                            "commonCoreStandardId": "",
                            "depthOfKnowledge": "",
                            "domain": "MATH",
                            "grade": 8,
                            "gradeLabel": "grade 3",
                            "interactionTypeCode": "",
                            "interactionTypeLabel": "",
                            "isPerformanceItem": false,
                            "itemKey": 1,
                            "subjectCode": "MATH",
                            "subjectLabel": "math",
                            "targetHash": 212,
                            "targetId": "23463467",
                            "targetShortName": "1-3",
                            "title": "",
                          },
                          "rubrics": Array [
                            Object {
                              "language": "english",
                              "rubricEntries": Array [
                                Object {
                                  "name": "",
                                  "scorepoint": "",
                                  "value": "",
                                },
                              ],
                              "samples": Array [
                                Object {
                                  "maxValue": "",
                                  "minValue": "",
                                  "sampleResponses": Array [
                                    Object {
                                      "name": "",
                                      "purpose": "",
                                      "sampleContent": "",
                                      "scorePoint": "",
                                    },
                                  ],
                                },
                              ],
                            },
                          ],
                          "targetDescription": "string",
                        },
                        "id": "12334",
                        "questionNumber": 1,
                        "view": Object {
                          "captured": true,
                          "html": null,
                          "id": "12334",
                          "picturePath": null,
                          "type": 1,
                        },
                      },
                    ],
                  }
                }
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "grade": "3",
              "subject": "math",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "12334",
            "nodeType": "class",
            "props": Object {
              "itemData": Object {
                "passage": null,
                "questions": Array [
                  Object {
                    "data": Object {
                      "associatedItems": "string",
                      "commonCoreStandardsDescription": "string",
                      "depthOfKnowledge": "string",
                      "educationalDifficulty": "string",
                      "evidenceStatement": "string",
                      "itemCardViewModel": Object {
                        "bankKey": 1,
                        "claimCode": "234536",
                        "claimLabel": "Math Claim",
                        "commonCoreStandardId": "",
                        "depthOfKnowledge": "",
                        "domain": "MATH",
                        "grade": 8,
                        "gradeLabel": "grade 3",
                        "interactionTypeCode": "",
                        "interactionTypeLabel": "",
                        "isPerformanceItem": false,
                        "itemKey": 1,
                        "subjectCode": "MATH",
                        "subjectLabel": "math",
                        "targetHash": 212,
                        "targetId": "23463467",
                        "targetShortName": "1-3",
                        "title": "",
                      },
                      "rubrics": Array [
                        Object {
                          "language": "english",
                          "rubricEntries": Array [
                            Object {
                              "name": "",
                              "scorepoint": "",
                              "value": "",
                            },
                          ],
                          "samples": Array [
                            Object {
                              "maxValue": "",
                              "minValue": "",
                              "sampleResponses": Array [
                                Object {
                                  "name": "",
                                  "purpose": "",
                                  "sampleContent": "",
                                  "scorePoint": "",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                      "targetDescription": "string",
                    },
                    "id": "12334",
                    "questionNumber": 1,
                    "view": Object {
                      "captured": true,
                      "html": null,
                      "id": "12334",
                      "picturePath": null,
                      "type": 1,
                    },
                  },
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "body",
      },
    ],
    "type": "html",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <head>
            <meta
              charSet="UTF-8"
            />
            <link
              href="http://example.com/style.css"
              rel="stylesheet"
            />
            <link
              href="http://example.com/Shared/CSS/Universal/items.css"
              rel="stylesheet"
            />
            <link
              href="http://example.com/Projects/SBAC/css/modernItems.css"
              rel="stylesheet"
            />
          </head>,
          <body>
            <CoverPage
              grade="3"
              subject="math"
            />
            <ItemViewContainer
              itemData={
                Object {
                  "passage": null,
                  "questions": Array [
                    Object {
                      "data": Object {
                        "associatedItems": "string",
                        "commonCoreStandardsDescription": "string",
                        "depthOfKnowledge": "string",
                        "educationalDifficulty": "string",
                        "evidenceStatement": "string",
                        "itemCardViewModel": Object {
                          "bankKey": 1,
                          "claimCode": "234536",
                          "claimLabel": "Math Claim",
                          "commonCoreStandardId": "",
                          "depthOfKnowledge": "",
                          "domain": "MATH",
                          "grade": 8,
                          "gradeLabel": "grade 3",
                          "interactionTypeCode": "",
                          "interactionTypeLabel": "",
                          "isPerformanceItem": false,
                          "itemKey": 1,
                          "subjectCode": "MATH",
                          "subjectLabel": "math",
                          "targetHash": 212,
                          "targetId": "23463467",
                          "targetShortName": "1-3",
                          "title": "",
                        },
                        "rubrics": Array [
                          Object {
                            "language": "english",
                            "rubricEntries": Array [
                              Object {
                                "name": "",
                                "scorepoint": "",
                                "value": "",
                              },
                            ],
                            "samples": Array [
                              Object {
                                "maxValue": "",
                                "minValue": "",
                                "sampleResponses": Array [
                                  Object {
                                    "name": "",
                                    "purpose": "",
                                    "sampleContent": "",
                                    "scorePoint": "",
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                        "targetDescription": "string",
                      },
                      "id": "12334",
                      "questionNumber": 1,
                      "view": Object {
                        "captured": true,
                        "html": null,
                        "id": "12334",
                        "picturePath": null,
                        "type": 1,
                      },
                    },
                  ],
                }
              }
            />
          </body>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <meta
                charSet="UTF-8"
              />,
              <link
                href="http://example.com/style.css"
                rel="stylesheet"
              />,
              <link
                href="http://example.com/Shared/CSS/Universal/items.css"
                rel="stylesheet"
              />,
              <link
                href="http://example.com/Projects/SBAC/css/modernItems.css"
                rel="stylesheet"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "charSet": "UTF-8",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "http://example.com/style.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "http://example.com/Shared/CSS/Universal/items.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "http://example.com/Projects/SBAC/css/modernItems.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
          ],
          "type": "head",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CoverPage
                grade="3"
                subject="math"
              />,
              Array [
                <ItemViewContainer
                  itemData={
                    Object {
                      "passage": null,
                      "questions": Array [
                        Object {
                          "data": Object {
                            "associatedItems": "string",
                            "commonCoreStandardsDescription": "string",
                            "depthOfKnowledge": "string",
                            "educationalDifficulty": "string",
                            "evidenceStatement": "string",
                            "itemCardViewModel": Object {
                              "bankKey": 1,
                              "claimCode": "234536",
                              "claimLabel": "Math Claim",
                              "commonCoreStandardId": "",
                              "depthOfKnowledge": "",
                              "domain": "MATH",
                              "grade": 8,
                              "gradeLabel": "grade 3",
                              "interactionTypeCode": "",
                              "interactionTypeLabel": "",
                              "isPerformanceItem": false,
                              "itemKey": 1,
                              "subjectCode": "MATH",
                              "subjectLabel": "math",
                              "targetHash": 212,
                              "targetId": "23463467",
                              "targetShortName": "1-3",
                              "title": "",
                            },
                            "rubrics": Array [
                              Object {
                                "language": "english",
                                "rubricEntries": Array [
                                  Object {
                                    "name": "",
                                    "scorepoint": "",
                                    "value": "",
                                  },
                                ],
                                "samples": Array [
                                  Object {
                                    "maxValue": "",
                                    "minValue": "",
                                    "sampleResponses": Array [
                                      Object {
                                        "name": "",
                                        "purpose": "",
                                        "sampleContent": "",
                                        "scorePoint": "",
                                      },
                                    ],
                                  },
                                ],
                              },
                            ],
                            "targetDescription": "string",
                          },
                          "id": "12334",
                          "questionNumber": 1,
                          "view": Object {
                            "captured": true,
                            "html": null,
                            "id": "12334",
                            "picturePath": null,
                            "type": 1,
                          },
                        },
                      ],
                    }
                  }
                />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "grade": "3",
                "subject": "math",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "12334",
              "nodeType": "class",
              "props": Object {
                "itemData": Object {
                  "passage": null,
                  "questions": Array [
                    Object {
                      "data": Object {
                        "associatedItems": "string",
                        "commonCoreStandardsDescription": "string",
                        "depthOfKnowledge": "string",
                        "educationalDifficulty": "string",
                        "evidenceStatement": "string",
                        "itemCardViewModel": Object {
                          "bankKey": 1,
                          "claimCode": "234536",
                          "claimLabel": "Math Claim",
                          "commonCoreStandardId": "",
                          "depthOfKnowledge": "",
                          "domain": "MATH",
                          "grade": 8,
                          "gradeLabel": "grade 3",
                          "interactionTypeCode": "",
                          "interactionTypeLabel": "",
                          "isPerformanceItem": false,
                          "itemKey": 1,
                          "subjectCode": "MATH",
                          "subjectLabel": "math",
                          "targetHash": 212,
                          "targetId": "23463467",
                          "targetShortName": "1-3",
                          "title": "",
                        },
                        "rubrics": Array [
                          Object {
                            "language": "english",
                            "rubricEntries": Array [
                              Object {
                                "name": "",
                                "scorepoint": "",
                                "value": "",
                              },
                            ],
                            "samples": Array [
                              Object {
                                "maxValue": "",
                                "minValue": "",
                                "sampleResponses": Array [
                                  Object {
                                    "name": "",
                                    "purpose": "",
                                    "sampleContent": "",
                                    "scorePoint": "",
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                        "targetDescription": "string",
                      },
                      "id": "12334",
                      "questionNumber": 1,
                      "view": Object {
                        "captured": true,
                        "html": null,
                        "id": "12334",
                        "picturePath": null,
                        "type": 1,
                      },
                    },
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "body",
        },
      ],
      "type": "html",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
